helix-core/src/main/java/org/apache/helix/controller/rebalancer/GlobalRebalancer.java

```diff
@@ -49,19 +52,29 @@ public RebalanceFailureReason(RebalanceFailureType type, String reason) {
       _reason = reason;
     }
 
-    public RebalanceFailureType get_type() {
+    public RebalanceFailureType getType() {
       return _type;
     }
 
-    public String get_reason() {
+    public String getReason() {
       return _reason;
     }
   }
 
-  void init(HelixManager manager);
-
-  Map<String, IdealState> computeNewIdealState(final CurrentStateOutput currentStateOutput,
-      T clusterData, Map<String, Resource> resourceMap);
+  /**
+   * Compute the new IdealStates for all the resources input. The IdealStates include both the new
+   * partition assignment (in the listFiles) and the new replica state mapping (in the mapFields).
+   *
+   * @param clusterData        The Cluster status data provider.
+   * @param resourceMap        A map containing all the rebalancing resources.
+   * @param currentStateOutput The present Current State of the cluster.
+   * @return A map containing the computed new IdealStates.
+   */
+  Map<String, IdealState> computeNewIdealStates(T clusterData, Map<String, Resource> resourceMap,
```

Comment:
If we'd like to differentiate between incremental and global computation, we can create 2 methods
```
Map<String, IdealState> computeIdealStatesForAllResources();
Map<String, IdealState> computeIdealStatesForResources();
```

Reply:
The design is that the calculation stage won't be aware of the difference. So better not to expose it.